featuretools/tests/primitive_tests/test_rolling_primitive.py

```diff
@@ -133,12 +214,113 @@ def test_rolling_count_primitive_min_periods_nans(min_periods, expected_num_nams
         (5, 4)
     ]
 )
-def test_rolling_count_with_no_gap(min_periods, expected_num_nams, rolling_series_pd):
-    window_length = 5
-    gap = 0
-
+@pytest.mark.parametrize("window_length, gap", [('5d', '0d'), (5, 0)])
+def test_rolling_count_with_no_gap(window_length, gap, min_periods, expected_num_nams, rolling_series_pd):
     primitive_instance = RollingCount(window_length=window_length, gap=gap, min_periods=min_periods)
     primitive_func = primitive_instance.get_function()
     vals = pd.Series(primitive_func(rolling_series_pd.index))
 
     assert vals.isna().sum() == expected_num_nams
+
+
+@pytest.mark.parametrize("primitive", [RollingCount, RollingMax, RollingMin, RollingMean])
+def test_rolling_primitives_non_uniform_data(primitive):
```

Comment:
Might be worth it to actually pre-compute the expected series and check with `assert_series_equal`? My reasoning is that in two weeks when we ask ourselves how this works, it'll be easier/faster to just look at the expected answer in the unit tests. 

That being said, I stepped through this manually and the resulting series matches what I'd expect. Right now, the window size in this test only includes one observation so all the primitives return the same result. I think it'd be good if we test on some situations where there is more than one observation per window.

Maybe something like this could work:

```python
import pandas as pd
from featuretools.primitives import RollingMean, RollingCount

# When the data isn't uniform, this impacts the number of values in each rolling window
datetimes = (list(pd.date_range(start='2017-01-01', freq='1d', periods=3)) +
             list(pd.date_range(start='2017-01-10', freq='2d', periods=4)) +
             list(pd.date_range(start='2017-01-22', freq='1d', periods=7)))

no_freq_series = pd.Series(range(len(datetimes)), index=datetimes)
```

Reply:
@freddyaboulton both good ideas! I added two tests in 4f90dd2bd94b094d8cce556e454990761fe9b9de that test the primitives on a series that has windows with multiple observations present and use `assert_series_equal`. If the tests from this commit were along the lines of what you were thinking, I'll go ahead and update the other tests.

Reply:
Also updated a test in `test_primitive_utils` to use the same approach for non uniform series: b0e5e25